### DISCUSSION TITLE
How is the better performance achieved with fewer parity disks?

### DISCUSSION THREAD

**Discussion by Clara12062** [minio/minio#18564] [Q&A] [answered]
As I change the configuration of storage_class, the write performance changes: the smaller the parity disks, the better the performance.

How is the better performance achieved with fewer parity disks?
I have three reasons, but I'm not sure what leads to better performance.
  1. As the number of data disks increases, each shard becomes smaller.
  2. As the number of data disks increases, the number of drivers to perform concurrent writes increases.
  3. As data disks increase, erasure coding becomes simpler and faster to compute.

**Comment by klauspost** [comment, marked as answer]
Points 1 & 2 only matter for big objects, where you hit throughput limits and not IOPS limits.

Point 3 doesn't matter. All erasure coding is bound by memory speed, so more/less doesn't make much, if any difference.

Less parity only affects storage *efficiency* and only has very little impact on speed.

**Reply by Clara12062** [reply]
Thanks for your fast reply!

Does storage *effiency* mean throughput and ops?

**Reply by klauspost** [reply]
No it means more usable space per byte of raw capacity. See https://min.io/product/erasure-code-calculator

**Reply by Clara12062** [reply]
ok, I see.

Thanks again for your reply.
